.pre-commit-config.yaml

```diff
@@ -0,0 +1,232 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+
+minimum_pre_commit_version: '2.8.2'
+
+default_language_version:
+  python: python3.7
+
+default_stages: [commit]
+
+exclude: '^example-site/themes/lektor-icon$|\.min\.js$|jquery\.stellar\.js$|icomoon\.css$|magnific-popup\.css$'
+
+
+repos:
+# Meta checks
+- repo: meta
+  hooks:
+  - id: check-useless-excludes
+
+# General fixers
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v3.3.0
+  hooks:
+  - id: trailing-whitespace
+    exclude_types: [svg]
+  - id: mixed-line-ending
+    name: Normalize mixed line endings
+    args: [--fix=lf]
+  - id: end-of-file-fixer
+    exclude_types: [svg]
+  - id: fix-byte-order-marker
+    name: Remove Unicode BOM
+
+# More general fixers
+- repo: https://github.com/Lucas-C/pre-commit-hooks
+  rev: v1.1.9
+  hooks:
+  - id: remove-tabs
+    name: Replace tabs with spaces
+    types: [text]
+    exclude_types: [svg]
+    args: [--whitespaces-count, '2']
```

Comment:
Shouldn't we replaced tabs for 4 spaces instead?

Reply:
Actually, 2 spaces is what most or all of the CSS, HTML, Jinja, JS, etc in both the theme and website repos have used, and is is the accepted standard for web development work, recommended by most style guides and the defaults for most of the corresponding tooling and style-standardization tools that are implemented here. We could change it, but it would create needless churn touching most of the lines of code in the repo without clear benefit, and it was simpler to just stick to one standard for everything in a given repo.

Also, keep in mind this fixer just does the tab->space replacement in generic files; per-language/file type formatters can easily be set to automatically apply any desired level of indendation if we so choose.

Reply:
Ok, no problem and thanks for the explanation.